src/Categories/Adjoint/Construction/Adjunctions/Properties.agda

```diff
@@ -0,0 +1,186 @@
+{-# OPTIONS --without-K  --allow-unsolved-metas #-}
+
+open import Level
+open import Categories.Category.Core using (Category)
+open import Categories.Category
+open import Categories.Monad
+
+module Categories.Adjoint.Construction.Adjunctions.Properties {o ℓ e} {C : Category (o ⊔ ℓ ⊔ e) (ℓ ⊔ e) e} (M : Monad C) where
+
+private
+  module C = Category C
+  module M = Monad M
+
+open import Categories.Adjoint using (Adjoint)
+open import Categories.Functor using (Functor)
+open import Categories.Morphism.Reasoning
+open import Categories.NaturalTransformation hiding (id)
+open NaturalTransformation using (η)
+open import Categories.NaturalTransformation.NaturalIsomorphism using (NaturalIsomorphism; niHelper; sym; associator)
+
+open import Categories.Adjoint.Construction.Adjunctions M
+
+open import Categories.Object.Terminal (Split M)
+open import Categories.Object.Initial (Split M)
+open import Categories.Category.Construction.EilenbergMoore
+open import Categories.Category.Construction.Kleisli
+open import Categories.Adjoint.Construction.Kleisli M as KL
+open import Categories.Adjoint.Construction.EilenbergMoore M as EM
+
+open import Categories.Tactic.Category
+
+-- EM-object : SplitObj
+-- EM-object = record
+--   { D = EilenbergMoore M
+--   ; F = EM.Free
+--   ; G = EM.Forgetful
+--   ; adj = EM.Free⊣Forgetful
+--   ; GF≃M = EM.FF≃F
+--   ; η-eq = begin
+--              M.F.₁ id ∘ M.η.η _ ≈⟨ M.F.identity ⟩∘⟨refl ⟩
+--              id ∘ M.η.η _       ≈⟨ identityˡ ⟩
+--              M.η.η _            ∎
+--   ; μ-eq = begin
+--              M.F.₁ id ∘ M.μ.η _                    ≈⟨ M.F.identity ⟩∘⟨refl ⟩
+--              id ∘ M.μ.η _                          ≈⟨ identityˡ ⟩
+--              M.μ.η _                               ≈⟨ Equiv.sym identityʳ ⟩
+--              M.μ.η _  ∘ id                         ≈⟨ refl⟩∘⟨ Equiv.sym M.F.identity ⟩
+--              M.μ.η _ ∘ M.F.₁ id                    ≈⟨ Equiv.sym identityʳ ⟩
+--              (M.μ.η _ ∘ M.F.₁ id) ∘ id             ≈⟨ assoc ⟩
+--              M.μ.η _ ∘ M.F.₁ id ∘ id               ≈⟨ refl⟩∘⟨  refl⟩∘⟨ Equiv.sym M.F.identity ⟩
+--              M.μ.η _ ∘ M.F.₁ id ∘ M.F.₁ id         ≈⟨ refl⟩∘⟨  refl⟩∘⟨ M.F.F-resp-≈ (Equiv.sym M.F.identity) ⟩
+--              M.μ.η _ ∘ M.F.₁ id ∘ M.F.₁ (M.F.₁ id) ∎
+--   } where open Category C
+--           open HomReasoning
+
+Kl-object : SplitObj
+Kl-object = record
+  { D = Kleisli M
+  ; F = KL.Free
+  ; G = KL.Forgetful
+  ; adj = KL.Free⊣Forgetful
+  ; GF≃M = KL.FF≃F
+  ; η-eq = begin
+             M.F.₁ id ∘ M.η.η _ ≈⟨ M.F.identity ⟩∘⟨refl ⟩
+             id ∘ M.η.η _       ≈⟨ identityˡ ⟩
+             M.η.η _            ∎
+  ; μ-eq = begin M.F.₁ id ∘ M.μ.η _                                     ≈⟨ M.F.identity ⟩∘⟨refl ⟩
```

Comment:
Agda doesn't like it when you put such long proofs inline. It's much better to name it, either in a private block above, or in the `where` clause below.

Also: `id-comm` and `id-comm-sym` exist for switching `id` from one side to the other in 1 line.

Reply:
Thank you, fixed! We found out about those helpers after writing those proofs...

Reply:
Great - in another pass, I'll see if I can find more simplifications. [I do my own proofs the same way, i.e. write them whichever way works, then compress/simplify by noticing the combinators that I ought to have used...]